Sprint-3/revise/implement/password-validator.test.js

```diff
@@ -14,65 +14,73 @@ To be valid, a password must:
 
 You must breakdown this problem in order to solve it. Find one test case first and get that working
 */
-const previousPasswords = ["Hello1@", "Samira2$", "Peter4*"]
-function isValidPassword(password){
-    
-    if(!password.length >=5){
-        return false;
-    }
-    var upperCaseLetters = /[A-Z]/g;
-    if(!password.match(upperCaseLetters)){
-        return false;
-    }
-    var lowerCaseLetters = /[a-z]/g;
-    if(!password.match(lowerCaseLetters)){
-        return false;
-    }
-    if(!password.match(/[0-9]/g)){
-        return false;
-    }
-    if(!password.match(/[@.#$!%^&*.?]/g)){
-        return false;
-    }
 
-    if(previousPasswords.includes(password)){
-        return false
-    }
-    return true;
-}
-
-
-describe("isValidPassword", () =>{
-    test('does not Have at least 5 characters.', () => {
-        expect(isValidPassword("lo5%")).toBe(false)
-    })
-
-    test('does not Have at least one English uppercase letter (A-Z)', () => {
-        expect(isValidPassword("hello5%")).toBe(false)
-    })
+function isValidPassword(password, previousPasswords = []) {
+  console.log(`password = ${password}`);
+  console.log("previousPasswords =", previousPasswords);
+  if (password.length < 5) {
+    return false;
+  }
+  const upperCaseLetters = /[A-Z]/g;
+  if (!password.match(upperCaseLetters)) {
+    return false;
+  }
+  const lowerCaseLetters = /[a-z]/g;
+  if (!password.match(lowerCaseLetters)) {
+    return false;
+  }
+  if (!password.match(/[0-9]/g)) {
+    return false;
+  }
+  if (!password.match(/[@.#$!%^&*.?]/g)) {
+    return false;
+  }
 
-    test('does not Have at least one English lowercase letter (a-z)', () => {
-        expect(isValidPassword("HELLO5%")).toBe(false)
-    })
-
-    test('does not Have at least one number', () => {
-        expect(isValidPassword("HELLO%")).toBe(false)
-    })
-
-    test('does not Have non-alphanumeric symbol ("!", "#", "$", "%", ".", "*", "&")', () => {
-        expect(isValidPassword("Hello5")).toBe(false)
-    })
+  if (previousPasswords.includes(password)) {
+    return false;
+  }
+  return true;
+}
+const previousPasswords = ["Hello1@", "Samira2$", "Peter4*"];
 
-    test.each(previousPasswords)('Must not be any previous password in the passwords array. ', (input) => {
-       console.log(input)
-        expect(isValidPassword(input.toUpperCase())).toBe(false)
-    })
+console.log(isValidPassword("Lo5%"));
 
-    test.each(["Heloothere5$", "Name7&", "Helloworld1."])('Must not be any previous password in the passwords array. ', (input) => {
-        console.log(input)
-         expect(isValidPassword(input)).toBe(true)
-     })
-     
-})
+describe("isValidPassword", () => {
```

Comment:
Could also keep the used password array in a variable, as in:
`const usedPasswords = ["Hello1@", "Samira2$", "Peter4*"];`
and then refer to the variable in the remaining part of the script.